types/es6-shim/index.d.ts

```diff
@@ -594,16 +593,16 @@ interface SetConstructor {
 
 declare var Set: SetConstructor;
 
-interface WeakMap<K extends object, V> {
+interface WeakMap<K extends Object, V> {
```

Comment:
This should be left as `object`. The `object` type was introduced in TypeScript 2.2 and has a different meaning than `Object`. We can use TypeScript 2.2 features in this file as it has been marked with a `// TypeScript Version: 2.2` comment at the top of the file.